src/overlays/actors/ovl_En_Aob_01/z_en_aob_01.c

```diff
@@ -885,40 +1050,48 @@ void func_809C2F34(EnAob01* this, PlayState* play) {
     this->actor.prevPos = this->actor.world.pos;
 }
 
-void func_809C2FA0(void) {
+/**
+ * Generates a randomized list of dog text offsets and stores them in weekEventRegs.
+ * EnDg uses these text offsets to determine what text to show when the player picks
+ * the dog up while equipped with the Mask of Truth, and EnRacedog uses them to
+ * determine what condition the dog is in.
+ */
+void EnAob01_InitializeDogTextOffsets(void) {
     u8 i;
     u8 rand;
-    u8 index;
-    u8 orig;
-    u8 orig2;
-    u8 sp44[7];
-    u8 sp34[] = {
+    u8 dogTextOffsetsTemp;
+    u8 packedDogTextOffsets[(RACEDOG_COUNT + 1) / 2];
+    u8 dogTextOffsets[RACEDOG_COUNT] = {
         0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,
     };
 
-    for (i = 0; i < ARRAY_COUNT(sp34); i++) {
-        rand = Rand_ZeroFloat(14.0f);
-        orig = sp34[i];
+    // Swap around the dog text offsets randomly 14 times.
+    for (i = 0; i < RACEDOG_COUNT; i++) {
+        rand = Rand_ZeroFloat(RACEDOG_COUNT);
 
-        sp34[i] = sp34[rand];
-        sp34[rand] = orig;
+        dogTextOffsetsTemp = dogTextOffsets[i];
+        dogTextOffsets[i] = dogTextOffsets[rand];
+        dogTextOffsets[rand] = dogTextOffsetsTemp;
     }
 
-    for (i = 0; i < ARRAY_COUNT(sp44); i++) {
-        gSaveContext.save.saveInfo.weekEventReg[42 + i] = 0;
-        sp44[i] = 0;
+    // Clear out all weekEventRegs related to the dog text offsets.
+    for (i = 0; i < ARRAY_COUNT(packedDogTextOffsets); i++) {
+        CLEAR_WEEKEVENTREG_DOG_RACE_TEXT_OFFSETS(i);
+        packedDogTextOffsets[i] = 0;
     }
 
-    for (i = 0; i < ARRAY_COUNT(sp34); i++) {
-        orig2 = sp34[i];
-        index = i / 2;
+    // Go through all the dog text offsets and pack them into weekEventRegs.
+    for (i = 0; i < RACEDOG_COUNT; i++) {
+        u8 dogTextOffsetsTemp = dogTextOffsets[i];
+        u8 j = i / 2;
 
         if (i % 2) {
-            sp44[index] |= orig2 << 0x4;
-            gSaveContext.save.saveInfo.weekEventReg[42 + index] =
-                ((void)0, gSaveContext.save.saveInfo.weekEventReg[42 + index]) | sp44[index];
+            //! @note sets text offsets for both even and odd entry.
+            //! Requires `RACEDOG_COUNT` to be even to work properly.
+            packedDogTextOffsets[j] |= dogTextOffsetsTemp << 4;
```

Comment:
I know some comments would be duplicated, but I think it's worth mentioning this comment at the definition of `RACEDOG_COUNT` (Requires `RACEDOG_COUNT` to be even to work properly.`)